main.go

```diff
@@ -311,6 +328,10 @@ func main() { //nolint:funlen,maintidx
 		setupLog.Error(err, "unable to set up ready check")
 		os.Exit(1)
 	}
+	if err := initComponents(platform); err != nil {
+		setupLog.Error(err, "unable to init components")
+		os.Exit(1)
```

Comment:
Good question. One one hand it makes problems obvious on startup. On it would make some of the system working even if dashboard tries to use wrong repo.

Reply:
what i am afarid is 
if compoent X fail to update, but user does not even want to enable component X but only Y and Z, should that make pod crash ?
in thoery , it should not even fail, 
and in another theory if it was a glitch: the restart of pod , due to crash, can fix X in the next run.

Reply:
If fatal init is not needed, implicit init() will work

Reply:
For non-fatal case I would like at least to log the error. Would it be ok to pass Context and use Bartosz' idea to wrap logger in it?

Reply:
I made it non-fatal, but on the per-Init level. Leaving possibility :)